3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js

```
@@ -17,29 +17,41 @@ Full description at: https://github.com/HackYourFuture/Homework/blob/main/3-Usin
    url with `.shx`. There is no server at the modified url, therefore this 
    should result in a network (DNS) error.
 ------------------------------------------------------------------------------*/
-function requestData(url) {
-  // TODO return a promise using `fetch()`
+async function requestData(url) {
+  try {
+    const response = await fetch(url);
+    if (response.ok) {
+      return await response.json();
+    }
+    renderError(`Error loading content occurs: ${response.statusText}`);
```

It is better for this function to be only to be concerned with fetching data. It should not render anything, not even an error. At present, you are violating the Separations of Concerns principle. Catching the error and rendering is best done by a higher-up function, in this case in function `main()`. You don't need a try/catch block here, just let the error bubble the call chain to function `main()`.

Here is how it was done in the prep exercise:

https://github.com/remarcmij/usingapis-qanda/blob/main/week2/pokemon/index.js#L4